packages/dappmanager/src/modules/ethClient/ethereumClient.ts

```diff
@@ -104,98 +110,158 @@ export class EthereumClient {
-    const currentEndpointConfig = await getDnCoreNetworkContainerConfig(
-      previousEthClientContainerName
-    );
+    const fullnodeAlias = network === "mainnet" ? params.FULLNODE_ALIAS : `${network}.${params.FULLNODE_ALIAS}`;
+
+    logs.info(`Updating fullnode alias (${fullnodeAlias}) for network ${network} from ${prevExecClientDnpName} to ${newExecClientDnpName}`);
+
+    if (prevExecClientDnpName === newExecClientDnpName) return;
+
+    if (prevExecClientDnpName) {
+      logs.info(`Removing fullnode alias (${fullnodeAlias}) from previous execution client for network ${network} (${prevExecClientDnpName})`);
+      await this.removeAliasFromPreviousExecClient({ execClientDnpName: prevExecClientDnpName, fullnodeAlias });
+    }
+
+    // New execution client can be undefined if the user sets remote Ethereum repository, for example
+    if (newExecClientDnpName) {
+      logs.info(`Adding fullnode alias (${fullnodeAlias}) to new execution client for network ${network} (${newExecClientDnpName})`);
+      this.addAliasToNewExecClient({ execClientDnpName: newExecClientDnpName, fullnodeAlias });
+    }
+
+  }
+
+  private async removeAliasFromPreviousExecClient<T extends Network>({ execClientDnpName, fullnodeAlias }: { execClientDnpName: ExecutionClient<T>, fullnodeAlias: string }): Promise<void> {
+    const execClientPkg = await packageGet({ dnpName: execClientDnpName });
+
+    const serviceName = this.getServiceNameFromPackage(execClientPkg);
+    const containerName = this.getContainerNameFromService(execClientPkg, serviceName);
+
+    await this.removeContainerAliasFromDockerNetwork({ containerName, fullnodeAlias });
+
+    this.removeFullnodeAliasFromCompose({ execClientDnpName, execClientServiceName: serviceName, fullnodeAlias });
+  }
+
+  private async addAliasToNewExecClient<T extends Network>({ execClientDnpName, fullnodeAlias }: { execClientDnpName: ExecutionClient<T>, fullnodeAlias: string }): Promise<void> {
+    const execClientPkg = await packageGet({ dnpName: execClientDnpName });
+
+    const serviceName = this.getServiceNameFromPackage(execClientPkg);
+    const containerName = this.getContainerNameFromService(execClientPkg, serviceName);
+
+    this.addContainerAliasToDockerNetwork({ containerName, fullnodeAlias });
+
+    this.addFullnodeAliasToCompose({ execClientDnpName, execClientServiceName: serviceName, fullnodeAlias });
+  }
+
+  private getServiceNameFromPackage(pkg: InstalledPackageDetailData): string {
+    return pkg.manifest?.mainService || pkg.containers[0].serviceName;
+  }
+
+  private getContainerNameFromService(pkg: InstalledPackageDetailData, serviceName: string): string {
+    return pkg.containers.find(container => container.serviceName === serviceName)?.containerName || pkg.containers[0].containerName;
+  }
+
+  private async removeContainerAliasFromDockerNetwork({ containerName, fullnodeAlias }: { containerName: string, fullnodeAlias: string }): Promise<void> {
```

Comment:
we could rename the parameter "fullnodeAlias" to "aliasToRemove" or something like that, right? Makes it clearer